notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/StreamBindingNotificationEventUtils.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.events;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import com.expediagroup.streamplatform.streamregistry.avro.AvroEvent;
+import com.expediagroup.streamplatform.streamregistry.avro.AvroKey;
+import com.expediagroup.streamplatform.streamregistry.avro.AvroKeyType;
+import com.expediagroup.streamplatform.streamregistry.avro.AvroStreamBinding;
+import com.expediagroup.streamplatform.streamregistry.model.Status;
+import com.expediagroup.streamplatform.streamregistry.model.StreamBinding;
+
+@Slf4j
+public class StreamBindingNotificationEventUtils {
+
+  public static AvroKey toAvroKeyRecord(StreamBinding streamBinding) {
+    validateStreamBindingKey(streamBinding);
+
+    val key = streamBinding.getKey();
+    val streamName = key.getStreamName();
+    val streamVersion = key.getStreamVersion();
+    val streamDomain = key.getStreamDomain();
+    val infrastructureName = key.getInfrastructureName();
+    val infrastructureZone = key.getInfrastructureZone();
+
+    var zoneKey = AvroKey.newBuilder()
+        .setId(infrastructureZone)
+        .setType(AvroKeyType.ZONE)
+        .build();
+
+    var infrastructureKey = AvroKey.newBuilder()
+        .setId(infrastructureName)
+        .setParent(zoneKey)
+        .setType(AvroKeyType.INFRASTRUCTURE)
+        .build();
+
+    var domainKey = AvroKey.newBuilder()
+        .setId(streamDomain)
+        .setParent(infrastructureKey)
+        .setType(AvroKeyType.DOMAIN)
+        .build();
+
+    var streamKey = AvroKey.newBuilder()
+        .setId(streamName)
+        .setParent(domainKey)
+        .setType(AvroKeyType.STREAM)
+        .build();
+
+    return AvroKey.newBuilder()
+        .setId(streamVersion.toString())
+        .setParent(streamKey)
```

Comment:
Hmm how to set the parent when it has lineage from two sides (stream and infrastructure)

Reply:
done

Reply:
done